### PR TITLE
feat: support more events in webhook parse in go-scm for gitness

### PR DESCRIPTION
**Pull request by abhinav-harness**

### CODE REVIEW

File: scm/driver/harness/webhook.go [hunk 1]
```diff
@@ -271,6 +275,7 @@ func convertPullReq(pr pullReq, ref ref, commit hookCommit) scm.PullRequest {
 		Closed: pr.State != "open",
```
Comment (rutvijmehta-harness) [inline scm/driver/harness/webhook.go:275]: Do we need `pr.State == "closed"`?

Comment (johannesHarness) [reply scm/driver/harness/webhook.go:275]: agree

File: scm/driver/harness/webhook.go [hunk 2]
```diff
@@ -271,6 +275,7 @@ func convertPullReq(pr pullReq, ref ref, commit hookCommit) scm.PullRequest {
...
 		Source: pr.SourceBranch,
 		Target: pr.TargetBranch,
+		Merged: pr.State == "merged",
 		Fork:   "fork",
```
Comment (johannesHarness) [inline scm/driver/harness/webhook.go:279]: should we keep this empty? not sure why this says "fork"

File: scm/driver/harness/webhook.go (hunk 1 above)
Comment (abhinav-harness) [reply scm/driver/harness/webhook.go:275]: pr merged is also pr closed, that is what we are doing in github's case too.

File: scm/driver/harness/webhook.go (hunk 2 above)
Comment (abhinav-harness) [reply scm/driver/harness/webhook.go:279]: don't want to change something which is working in this PR ⛔

File: scm/driver/harness/webhook.go (hunk 1 above)
Comment (abhinav-harness) [reply scm/driver/harness/webhook.go:275]: can change it.

**Review by johannesHarness — APPROVED** [review]

**Review by rutvijmehta-harness — APPROVED** [review]

**Pull request merged**